src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+  let newState = JSON.parse(JSON.stringify(state));
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(newState, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
I think after the cases, there should be `default:` with an error just like in the previous task, but I'm not sure. ;)

Reply:
yes, @Matylda-Lewandowska is right, default is required 😉